pandas/tests/io/test_pytables.py

```diff
@@ -5007,8 +5007,9 @@ def test_read_hdf_errors(self):
             store = HDFStore(path, mode='r')
             store.close()
             pytest.raises(IOError, read_hdf, store, 'df')
-            with open(path, mode='r') as store:
-                pytest.raises(NotImplementedError, read_hdf, store, 'df')
+
+    def test_read_hdf_generic_buffer_errors(self):
+        pytest.raises(NotImplementedError, read_hdf, BytesIO(b''), 'df')
```

Comment:
@jreback I had to modify this test to hit the block [here](https://github.com/pandas-dev/pandas/pull/16301/files#diff-1b15d1477da3a0548d2dd72a5d023d00R321). I'm pretty sure you were just using `store` as an example of a buffer that isn't an HDFStore or filepath. But with these changes it'll be coerced to a path.